eclair-core/src/test/scala/fr/acinq/eclair/integration/PerformanceIntegrationSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.integration
+
+import akka.testkit.TestProbe
+import com.typesafe.config.ConfigFactory
+import fr.acinq.bitcoin.SatoshiLong
+import fr.acinq.eclair.MilliSatoshiLong
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.payment._
+import fr.acinq.eclair.payment.receive.MultiPartHandler.ReceivePayment
+import fr.acinq.eclair.payment.send.MultiPartPaymentLifecycle.PreimageReceived
+import fr.acinq.eclair.payment.send.PaymentInitiator
+import fr.acinq.eclair.router.Router
+
+import java.util.UUID
+import java.util.concurrent.Executors
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
+
+/**
+ * Created by PM on 15/03/2017.
+ */
+
+class PerformanceIntegrationSpec extends IntegrationSpec {
+
+  test("start eclair nodes") {
+    val commonPerfTestConfig = ConfigFactory.parseMap(Map(
+      "eclair.max-funding-satoshis" -> 100_000_000,
+      "eclair.max-accepted-htlcs" -> 30,
+      "eclair.file-backup.enabled" -> false,
+      "eclair.db.sqlite.mode" -> "wal",
+      "eclair.db.sqlite.sync" -> "full",
+    ).asJava)
+
+    instantiateEclairNode("A", ConfigFactory.parseMap(Map("eclair.node-alias" -> "A", "eclair.server.port" -> 29730).asJava).withFallback(commonPerfTestConfig).withFallback(commonFeatures).withFallback(commonConfig)) // A's channels are private
+    instantiateEclairNode("B", ConfigFactory.parseMap(Map("eclair.node-alias" -> "B", "eclair.server.port" -> 29731).asJava).withFallback(commonPerfTestConfig).withFallback(commonFeatures).withFallback(commonConfig))
+  }
+
+  test("connect nodes") {
+    // A---B
+
+    val eventListener = TestProbe()
+    nodes.values.foreach(_.system.eventStream.subscribe(eventListener.ref, classOf[ChannelStateChanged]))
+
+    connect(nodes("A"), nodes("B"), 100_000_000 sat, 0 msat)
+
+    // confirming the funding tx
+    generateBlocks(6)
+
+    within(60 seconds) {
+      eventListener.expectMsgType[ChannelStateChanged](60 seconds).currentState == NORMAL
+    }
+  }
+
+  test("wait for channels balance") {
```

Comment:
The name is a bit misleading, shouldn't it just be:
```suggestion
  test("wait for channels") {
```